### PR TITLE
HIVE-25438 : Update partition column stats fails with invalid syntax error for MySql

### PR DESCRIPTION
**Pull request by maheshk114**
…

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java [hunk 1]
```diff
@@ -647,6 +653,12 @@ public long getNextCSIdForMPartitionColumnStatistics(long numStats) throws MetaE
       jdoConn = pm.getDataStoreConnection();
       dbConn = (Connection) (jdoConn.getNativeConnection());
 
+      if (sqlGenerator.getDbProduct().isMYSQL()) {
+        try (Statement stmt = dbConn.createStatement()) {
+          stmt.execute("SET @@session.sql_mode=ANSI_QUOTES");
+        }
+      }
+
```
Comment (ramesh0201) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java:661]: Can we move the two code blocks to a function and invoke the function in the two places.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java [hunk 2]
```diff
@@ -647,6 +653,12 @@ public long getNextCSIdForMPartitionColumnStatistics(long numStats) throws MetaE
       jdoConn = pm.getDataStoreConnection();
       dbConn = (Connection) (jdoConn.getNativeConnection());
 
+      if (sqlGenerator.getDbProduct().isMYSQL()) {
```
Comment (ramesh0201) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java:656]: Is there any way to filter and set only if the query contains quotes?

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java (hunk 1 above)
Comment (maheshk114) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java:661]: done

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java (hunk 2 above)
Comment (maheshk114) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java:656]: that will be costlier than setting the value in DB. As we dont know how many sql statements are getting executed in side a connection.

**Comment by ramesh0201** [comment]
+1. LGTM. Thanks for the patch!

**Pull request merged**